static/js/src/advantage/react/hooks/useUserSubscriptions.ts

```diff
@@ -30,6 +33,31 @@ export const selectFreeSubscription = (subscriptions: UserSubscription[]) =>
     (subscription) => subscription.type === UserSubscriptionType.Free
   ) || null;
 
+/**
+ * Whether there is a subscription with the provided status.
+ */
+const hasStatus = (
+  subscriptions: UserSubscription[],
+  status: keyof UserSubscriptionStatuses
+) => subscriptions.some(({ statuses }) => statuses[status]);
+
+/**
+ * Get a summary of statuses for all subscriptions.
+ */
+export const selectStatusesSummary = (
```

Comment:
As a side note, I find it slightly confusing that these functions are placed in `hooks` directory as technically they're not hook but selector functions.

Reply:
Yeah I wasn't quite sure where to put them. They're designed to work with the React Query select property (I can't link directly to the select docs, but it's on this page: https://react-query.tanstack.com/reference/useQuery) so I left them in the same file as the hook they're used with.

They're used like this: https://github.com/canonical-web-and-design/ubuntu.com/blob/ua-subscriptions/static/js/src/advantage/react/components/Subscriptions/SubscriptionList/SubscriptionList.tsx#L29

Maybe it would make sense to create a dir for API endpoint with a hook and selector file.